### PR TITLE
Preserve docstrings in auto-tracing

### PR DESCRIPTION
**Pull request by alexmojaki**
Closes https://github.com/pydantic/logfire/issues/440

### CODE REVIEW

File: logfire/_internal/ast_utils.py [hunk 1]
```diff
@@ -68,21 +68,36 @@ def rewrite_function(self, node: ast.FunctionDef | ast.AsyncFunctionDef, qualnam
         # Replace the body of the function with:
         #     with <logfire_method_call_node>:
         #         <original body>
+        body = node.body.copy()
+        new_body: list[ast.stmt] = []
+        if (
+            body
+            and isinstance(body[0], ast.Expr)
```
Comment (davidhewitt) [inline logfire/_internal/ast_utils.py:75]: I suppose it is guaranteed that `0` must exist (even if it is just `pass` or `...`)?

**Comment by davidhewitt** [comment]
All looks good to me, sorry for the late review 👍

File: logfire/_internal/ast_utils.py (hunk 1 above)
Comment (alexmojaki) [reply logfire/_internal/ast_utils.py:75]: I'm pretty sure it is, but I started the condition with `body and` just in case.

Comment (davidhewitt) [reply logfire/_internal/ast_utils.py:75]: Ah right, at this point Rust's lack of "empty collection is not truthy" has eroded this from my mind 😂

**Pull request merged**